pkg/git.go

```diff
@@ -171,11 +172,22 @@ func (p *GitPackage) Install(ctx context.Context, name, dir, version string) (st
 	// the ETag header included in the response.
 	isGitHubRemote, err := regexp.MatchString(`^(https|ssh)://github\.com/.+$`, p.Source.Remote)
 	if isGitHubRemote {
-		archiveUrl := fmt.Sprintf("%s/archive/%s.tar.gz", p.Source.Remote, version)
+		// Let git ls-remote decide if "version" is a ref or a SHA1 in the unlikely
+		// but possible event that a ref is exactly 40 hex characters
+		commitSha, err := remoteResolveRef(ctx, p.Source.Remote, version)
+
+		// If the ref resolution failed and "version" looks like a SHA1, assume it is one
+		// and proceed.
+		commitShaPattern, _ := regexp.Compile("^([0-9a-f]{40})$")
```

Comment:
As you want to ignore the error, you can use:
```suggestion
		commitShaPattern := regexp.MustCompile("^([0-9a-f]{40})$")
```

Reply:
Again, this is not sha256 compatible, can we relax this regex ? `version` is computed from: 
* the lock file
* provided by the user ([validated from liberal regexps](https://github.com/jsonnet-bundler/jsonnet-bundle)).

Reply:
> this is not sha256 compatible

Which version of git and which repository supports sha256 today?

Reply:
As you delegate to Git for this, I see an opportunity to not encode the implementation detail. But maybe this is too early.

Reply:
My understanding is that Git plumbing commands (such as this one) were designed for their output to be parsed and used for scripting. Assuming the sha256 transition plan changes this plumbing command to output a longer sha but keep everything else the same, we can relax the parsing.